src/Trace/Trace.php

```diff
@@ -0,0 +1,211 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Trace;
+
+use Google\Cloud\Core\Exception\NotFoundException;
+use Google\Cloud\Core\ValidateTrait;
+use Google\Cloud\Trace\Connection\ConnectionInterface;
+
+/**
+ * This plain PHP class represents a Trace resource. Traces belong to a
+ * project and have many TraceSpans. See:
+ * https://cloud.google.com/trace/docs/reference/v1/rest/v1/projects.traces#resource-trace
+ */
+class Trace
+{
+    use ValidateTrait;
+
+    /**
+     * @var ConnectionInterface Represents a connection to Stackdriver Trace.
+     */
+    private $connection;
+
+    /**
+     * The id of the project this trace belongs to.
+     * @var string
+     */
+    private $projectId;
+
+    /**
+     * The trace id for this trace. 128-bit numeric formatted as a 32-byte hex string
+     * @var string
+     */
+    private $traceId;
+
+    /**
+     * List of TraceSpans to report
+     * @var TraceSpan[]
+     */
+    private $spans = [];
+
+    /**
+     * Instantiate a new Trace instance.
+     *
+     * @param ConnectionInterface $connection The connection to Stackdriver Trace.
+     * @param string $projectId The id of the project this trace belongs to.
+     * @param string $traceId [optional] The id of the trace. If not provided, one will be generated
+     *        automatically for you.
+     * @param array $spans [optional] Array of TraceSpan constructor arguments. See
+     *        {@see Google\Cloud\Trace\TraceSpan::__construct()} for configuration details.
+     * }
+     */
+    public function __construct(ConnectionInterface $connection, $projectId, $traceId = null, $spans = null)
+    {
+        $this->connection = $connection;
+        $this->projectId = $projectId;
+        $this->traceId = $traceId ?: $this->generateTraceId();
+        if ($spans) {
+            $this->spans = array_map(function ($span) {
+                return new TraceSpan($span);
+            }, $spans);
+        }
+    }
+
+    /**
+     * Retrieves the trace's id.
+     *
+     * @return string
+     */
+    public function traceId()
+    {
+        return $this->traceId;
+    }
+
+    /**
+     * Retrieves the trace's projectId.
+     *
+     * @return string
+     */
+    public function projectId()
+    {
+        return $this->projectId;
+    }
+
+    /**
+     * Set the trace's projectId
+     *
+     * @param string $projectId
+     */
+    public function setProjectId($projectId)
+    {
+        $this->projectId = $projectId;
+    }
+
+    /**
+     * Returns a serializable array representing this trace. If no span data
+     * is cached, a network request will be made to retrieve it.
+     *
+     * @return array
+     */
+    public function info()
+    {
+        if (!$this->spans) {
+            $this->reload();
+        }
+
+        return [
```

Comment:
I think I'd prefer to keep this method (and `reload()) a bit closer to the idioms used elsewhere. Something like this:

```php
<?php

class Trace
{
    private $traceId;
    private $projectId;
    private $spans;
    private $info;

    public function info(array $options = [])
    {
        return $this->info ?: $this->reload($options);
    }

    public function reload(array $options = [])
    {
        $trace = $this->connection->getTrace([
            'projectId' => $this->projectId,
            'traceId' => $this->traceId
        ]);

        if (empty($trace)) {
            throw new NotFoundException('Trace ID does not exist', 404);
        }

        $this->spans = array_map(function ($span) {
            return new TraceSpan($span);
        }, $trace['spans']);

        return $this->info = $trace;
    }
}
```

For precedent, see `Google\Cloud\PubSub\Topic::info()`, `Google\Cloud\Logging\Sink::info()` and others.

This has the added benefit of not needing the `array_map`.